ObjectMapper/Core/Mapper.swift

```diff
@@ -102,6 +102,13 @@ public final class Mapper<N: BaseMappable> {
 			}
 		}
 
+		if let klass = N.self as? ImmutableMappable.Type {
+			if let maybeObject = try? klass.init(map: map) as? N, var object = maybeObject {
```

Comment:
I think I prefer approach number one. I think that if you define an model as `ImmutableMappable` you expect a error to be thrown if something was not correctly map.

With that said, I think it makes sense to add `mapArray` to the `ImmutableMappable` `Mapper` extension because I think this should be a `throwing` method. Yes, it will cause some code duplication, but I think it will provide a better interface for developers.